public/js/NeighbourPlot.js

```diff
@@ -66,109 +66,218 @@ function NeighbourPlot(sampleData, element) {
  *
  * @this {NeighbourPlot}
  */
-NeighbourPlot.prototype.drawScatterplot = function() {
+NeighbourPlot.prototype.drawScatterplot = function(reductionType) {
+    this.reduction_type = reductionType || 'tsne';
+
+    // find scale for dataset
+    this.setScale();
+
+    // draw a new canvas if it does not exist yet
+    if(!this.svg) {
+        this.svg = d3.select(this.element).append('svg')
+            .attr('width', this.fullWidth)
+            .attr('height', this.fullHeight)
+            .append('g')
+            .attr('transform', 'translate(' + this.margin.left + ',' + this.margin.top + ')');
+    }
+
+
+    // define tooltip function and append to SVG
+    if(!this.tip) {
+        this.tip = d3.tip()
+            .attr('class', 'd3-tip')
+            .offset([-10, 0])
+            .html(function(d) {
+                return "<p><strong>ID: </strong>" + d._id + "</span></p>" +
+                    "<p><strong>Gene: </strong>" + d.gene_name + "</span></p>";
```

Comment:
Ohh wait, the bold text is already done in the CSS! I'll just remove the redundant markup in the tooltip :)